datafusion/functions-array/src/udf.rs

```diff
@@ -166,146 +166,6 @@ impl ScalarUDFImpl for StringToArray {
     }
 }
 
-make_udf_function!(
-    Range,
-    range,
-    start stop step,
-    "create a list of values in the range between start and stop",
-    range_udf
-);
-#[derive(Debug)]
-pub struct Range {
-    signature: Signature,
-    aliases: Vec<String>,
-}
-impl Range {
-    pub fn new() -> Self {
-        use DataType::*;
-        Self {
-            signature: Signature::one_of(
-                vec![
-                    TypeSignature::Exact(vec![Int64]),
-                    TypeSignature::Exact(vec![Int64, Int64]),
-                    TypeSignature::Exact(vec![Int64, Int64, Int64]),
-                    TypeSignature::Exact(vec![Date32, Date32, Interval(MonthDayNano)]),
-                ],
-                Volatility::Immutable,
-            ),
-            aliases: vec![String::from("range")],
-        }
-    }
-}
-impl ScalarUDFImpl for Range {
-    fn as_any(&self) -> &dyn Any {
-        self
-    }
-    fn name(&self) -> &str {
-        "range"
-    }
-
-    fn signature(&self) -> &Signature {
-        &self.signature
-    }
-
-    fn return_type(&self, arg_types: &[DataType]) -> Result<DataType> {
-        use DataType::*;
-        Ok(List(Arc::new(Field::new(
-            "item",
-            arg_types[0].clone(),
-            true,
-        ))))
-    }
-
-    fn invoke(&self, args: &[ColumnarValue]) -> Result<ColumnarValue> {
-        let args = ColumnarValue::values_to_arrays(args)?;
-        match args[0].data_type() {
-            arrow::datatypes::DataType::Int64 => {
-                crate::kernels::gen_range(&args, false).map(ColumnarValue::Array)
-            }
-            arrow::datatypes::DataType::Date32 => {
-                crate::kernels::gen_range_date(&args, false).map(ColumnarValue::Array)
-            }
-            _ => {
-                exec_err!("unsupported type for range")
-            }
-        }
-    }
-
-    fn aliases(&self) -> &[String] {
-        &self.aliases
-    }
-}
-
-make_udf_function!(
```

Comment:
Likewise this looks like an accidental copy too -- here is the other copy:

https://github.com/apache/datafusion/blob/0ea9bc659cbcdd1d70cf8ea8c284634d65171c33/datafusion/functions-array/src/range.rs#L102-L163